### PR TITLE
feat: generate arbitrary mock tx with signature

### PR DESCRIPTION
**Pull request by eugypalu**
Resolves: #1322 

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [x] Testing

# What is the new behavior?

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/eth_provider/database/types/transaction.rs [hunk 1]
```diff
@@ -168,6 +172,7 @@ impl<'a> StoredTransaction {
 
         let transaction = Transaction {
             hash,
+            from: Address::arbitrary(u)?,
```
Comment (greged93) [inline src/eth_provider/database/types/transaction.rs:175]: you should probably extract `from` from the random Private key

Comment (eugypalu) [reply src/eth_provider/database/types/transaction.rs]: done, thanks

**Review by greged93 — DISMISSED** [review]

**Comment by greged93** [comment]
@eugypalu can you just rebase this?

**Review by greged93 — APPROVED** [review]

**Comment by tcoratger** [comment]
@eugypalu I think this is almost good. I wanted to replace our hardcoded values in a follow PR. Maybe there is something that I don't get but this test should pass no?

```rust
#[test]
    fn random_tx_signature() {
        let mut bytes = [0u8; 1024];
        rand::thread_rng().fill(bytes.as_mut_slice());

        let tx = StoredTransaction::arbitrary_with_optional_fields(&mut arbitrary::Unstructured::new(&bytes)).unwrap();

        let signature = reth_primitives::Signature {
            r: tx.signature.unwrap().r,
            s: tx.signature.unwrap().s,
            odd_y_parity: tx.signature.unwrap().y_parity.unwrap_or(reth_rpc_types::Parity(false)).0,
        };

        assert_eq!(tx.from, signature.recover_signer(tx.hash).unwrap());
    }
```

At the moment it fails for me, I'm not sure that

```rust
let signature = signer.sign_hash_sync(&signing_hash).map_err(|_| arbitrary::Error::IncorrectFormat)?;
```

is correct for signing the transaction, what do you think? Can we add this test to the file? 

Can't we use an approach like https://github.com/paradigmxyz/reth/blob/2840a957853be7bf78f4cacd96bcc2ff7c6b6aee/crates/e2e-test-utils/src/transaction.rs#L50-L51 with a Tokio `block_on` to make it sync or something like that with:

```rust
let mut primitive_transaction = reth_rpc_types::TransactionRequest {
            from: Some(Address::arbitrary(u)?),
            to: Some(reth_primitives::TxKind::Call(Address::arbitrary(u)?)),
            gas_price: Some(u64::arbitrary(u)? as u128),
            max_fee_per_gas: Some(u64::arbitrary(u)? as u128),
            max_priority_fee_per_gas: Some(u64::arbitrary(u)? as u128),
            gas: Some(u64::arbitrary(u)? as u128),
            value: Some(U256::arbitrary(u)?),
            input: reth_rpc_types::TransactionInput::new(reth_primitives::Bytes::arbitrary(u)?),
            nonce: Some(u64::arbitrary(u)?),
            chain_id: Some(u64::arbitrary(u)?),
            access_list: Some(reth_rpc_types::AccessList::arbitrary(u)?),
            max_fee_per_blob_gas: Some(u64::arbitrary(u)? as u128),
            blob_versioned_hashes: Some(vec![]),
            transaction_type: Some(u8::arbitrary(u)? % 3),
            sidecar: None,
        };

        let wallet = PrivateKeySigner::random();

        let signer = EthereumWallet::from(wallet);
        tx.build(&signer).await.unwrap();
```

**Comment by greged93** [comment]
I tried this Thomas and couldn't get the correct trait bounds for it to work, maybe I was wrong somewhere

**Pull request merged**